development/gcbuild/main.go

```diff
@@ -0,0 +1,318 @@
+}
+
+// validateOptions handles options validation. All checks should be provided here
+func validateOptions(o options) error {
+	// TODO(Ressetkk): These checks are great candidate to be in checks.go as separate check
+	var errs []error
+
+	if o.cloudbuild == "" {
+		errs = append(errs, fmt.Errorf("'cloudbuild' option is missing, please define this option using '--cloudbuild-file' flag"))
+	}
+	if o.buildDir == "" {
+		errs = append(errs, fmt.Errorf("'buildDir' option is missing, please define this option using '--build-dir' flag"))
+	}
+	if o.Project == "" {
+		errs = append(errs, fmt.Errorf("'project' option is missing in the config file"))
+	}
+	return errutil.NewAggregate(errs)
+}
+
+// checkDependencies checks if required binaries are present in PATH
+// If they are not present, then returns proper error message.
+func checkDependencies() error {
+	deps := []string{
+		"gcloud",
+		"gsutil",
+		"git",
+	}
+	var errs []error
+	for _, c := range deps {
+		if _, err := exec.LookPath(c); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return errutil.NewAggregate(errs)
+}
+
+func (o *options) gatherOptions(fs *flag.FlagSet) *flag.FlagSet {
+	fs.BoolVar(&o.silent, "silent", false, "Do not push build logs to stdout")
+	fs.StringVar(&o.configPath, "config", "", "Path to application config file")
+	fs.StringVar(&o.buildDir, "build-dir", ".", "Path to build directory")
+	fs.StringVar(&o.cloudbuild, "cloudbuild-file", "cloudbuild.yaml", "Path to cloudbuild.yaml file relative to build-dir")
+	fs.StringVar(&o.variant, "variant", "", "If variants.yaml file is present, define which variant should be built. If variants.yaml is not present, this flag will be ignored")
+	fs.StringVar(&o.logDir, "log-dir", "/logs/artifacts", "Path to logs directory where GCB logs will be stored")
+	// (Ressetkk): Maybe treat these flags as overrides for config file?
+	//fs.StringVar(&o.devRegistry, "dev-registry", "", "Registry URL where development/dirty images should land. If not set then the default registry is used. This flag is only valid when running in CI (CI env variable is set to `true`)")
+	//fs.StringVar(&o.project, "project", "", "GCP project name where build jobs will run")
+	//fs.StringVar(&o.stagingBucket, "staging-bucket", "", "Full name to the Google Cloud Storage bucket, where the source will be pushed beforehand. If not set, rely on Google Cloud Build")
+	//fs.StringVar(&o.logsBucket, "logs-bucket", "", "Full name to the Google Cloud Storage bucket, where the logs will be pushed after build finishes. If not set, rely on Google Cloud Build")
+	return fs
+}
+
+func main() {
+	if err := checkDependencies(); err != nil {
+		panic(err)
+	}
+
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	o := options{isCI: os.Getenv("CI") == "true"}
+	o.gatherOptions(fs)
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		panic(err)
+	}
+
+	c, err := ioutil.ReadFile(o.configPath)
+	if err != nil {
+		panic(err)
+	}
+
+	if err := o.ParseConfig(c); err != nil {
```

Comment:
I've added a strict check if config flag is missing.

Reply:
Can we inform user about such problem earlier not when the tool is executed to do its job? Same as other configuration we discussed?

Reply:
Not really I'm afraid. It's a program's flag which we are supposed to provide. If we plan on providing the tool, so the devs maintain the build jobs by themselves we will have to provide in the set-up instruction that this flag is mandatory and by default should be set to specific value.

Reply:
What about testing a prowjob definition if it provide this flag?

Reply:
Can we provide a default value for this flag?

Reply:
Sure, but how would you distinct a ProwJobs if they have different names? That'd work if we were the ones that maintain the jobs and not distribute them as in-repo jobs.

Let's tackle the problem in the follow-ups after the flow is at least kick-started, because testing prowjobs against some infra-required stuff is much broader topic.  Right now there is a fail if the config is not provided. With proper set-up docs this should be enough.

Reply:
You can check what command and what args a prowjob is using.

Reply:
As I asked, let's discuss it in the follow-ups. Current approach should be well enough for kick start and move forward with SLC-29.

Reply:
Which follow up I should use for this?

Reply:
Ideally #5835 but I see it as much broader topic that tackles testing of ALL prowjobs against some corporate measures, refactor of `development/jobs/tests` stuff (#4889). So I would mention this issue, but not set it as hard requirement for now.